### ISSUE TITLE
gRPC update to 1.30.0

### ISSUE THREAD

**Issue by Dave-Lowndes** [microsoft/vcpkg#12122]
Library name: gRPC

New version number: 1.30.0

**Comment by Maju-Ketchup** [comment]
I started work [here](https://github.com/Maju-Ketchup/vcpkg/tree/update/grpc/1.31) on an updated port.
Actually I'm new to Open Source projects and espacially vcpkg and don't know, if it is right what I am doing.
I try to fix the patches and added re2 as dependency. Also I try to go to version 1.31.x
In the moment, I get some linking issues for upb during the build and try to figure out where the problem is.

**Comment by Maju-Ketchup** [comment]
I'm still stuck with upb poblems.
Since @ras0219-msft and @JackBoosY did the patches for this, maybe they can help me understand what they did. 
An ELI5 would help.
I noticed in the logs that the linking to upb is broken in some parts.
The first error in the log shows:
`[...]\upb\upb\decode.c(44): error C2065: 'UPB_MAPTYPE_STRING': undeclared identifier`

**Comment by JackBoosY** [comment]
@Maju-Ketchup Could you open a PR to commit your changes?
That I can help you solve these problems.
